### PR TITLE
Add support for Exporting API and RP support for MultiHostname

### PR DESCRIPTION
**Pull request by solankisamir**
- Add Client for Exporting API as links
- Update RP support for default SslBinding and NegotiateClientCertificate in Custom Hostnames

Link to Spec https://github.com/Azure/azure-rest-api-specs/blob/current/specification/apimanagement/resource-manager/readme.md using the cmdline for 2017-03-01 version.

We have not yet published our nuget, waiting on some additional changes to the swagger spec.

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
Missing Rest spec PR
Missing version update for the new version.

**Comment by solankisamir** [comment]
@shahabhijeet added link to the spec. We used version 2017-03-01. Also, we have not published our nuget, hence did not think of updating the revision. Is that still required?

**Comment by markcowl** [comment]
@solankisamir  Yes, you should update to makr the difference from the previously published SDK *unless you didn't publish the source that is currently in the repo)

**Comment by solankisamir** [comment]
@markcowl I have not published the source in the repo.

File: src/SDKs/_metadata/apimanagement_resource-manager.txt [hunk 1]
```diff
@@ -1,10 +0,0 @@
-2017-08-01 15:52:08 UTC
```
Comment (markcowl) [inline src/SDKs/_metadata/apimanagement_resource-manager.txt]: Please don't remove this.  This file is used to track the metadata used to create your package.  When you run generate.cmd, it should replace this file for you

**Review by markcowl — CHANGES_REQUESTED** [review]
You need to update the version (your latest released version is 3.4.0-preview, so this shoudl be greater than that, and you need to include the metadata file, as specified in comments

**Comment by shahabhijeet** [comment]
@solankisamir can you help me understand what this version is
https://www.nuget.org/packages/Microsoft.Azure.Management.ApiManagement/3.4.0-preview
Are these two separate nuget packages?

**Comment by shahabhijeet** [comment]
@solankisamir please reopen the same PR once you have addressed all the feedback/questions.
Leaving a PR open for more than a week unaddressed affects our SLA data.

**Pull request closed without merging**